controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/JobRunner.java

```diff
@@ -34,18 +34,21 @@
     private final JobController jobs;
     private final ExecutorService executors;
     private final StepRunner runner;
+    private final Clock clock;
 
-    public JobRunner(Controller controller, Duration duration, JobControl jobControl) {
-        this(controller, duration, jobControl, Executors.newFixedThreadPool(32), new InternalStepRunner(controller));
+    public JobRunner(Controller controller, Duration duration, JobControl jobControl, Clock clock) {
+        this(controller, duration, jobControl, Executors.newFixedThreadPool(32), new InternalStepRunner(controller), clock);
     }
 
     @TestOnly
-    public JobRunner(Controller controller, Duration duration, JobControl jobControl, ExecutorService executors, StepRunner runner) {
+    public JobRunner(Controller controller, Duration duration, JobControl jobControl, ExecutorService executors,
+                     StepRunner runner, Clock clock) {
         super(controller, duration, jobControl);
         this.jobs = controller.jobController();
         this.jobs.setRunner(this::advance);
         this.executors = executors;
         this.runner = runner;
+        this.clock = clock;
```

Comment:
All `Maintainer`s have access to the controller's clock through `controller().clock()` already.

Reply:
Fixed